### PR TITLE
Allow bindings conformance tests for uninitialized components

### PR DESCRIPTION
**Pull request by mcandeia**
Signed-off-by: Marcos Candeia <marrcooos@gmail.com>

# Description

the bindings conformance tests are now considering the target operations before running the ping for the respective bindings components.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2347 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by artursouza — DISMISSED** [review]

File: tests/conformance/bindings/bindings.go [hunk 1]
```diff
@@ -143,23 +143,27 @@ func ConformanceTests(t *testing.T, props map[string]string, inputBinding bindin
...
+			// TODO: Ideally, all stable components should implenment ping function,
+			// so will only assert assert.Nil(t, err) finally, i.e. when current implementation
+			// implements ping in existing stable components
+			if errOut != nil {
+				assert.EqualError(t, errOut, "ping is not implemented by this output binding")
+			} else {
+				assert.Nil(t, errOut)
+			}
 		}
 	})
```
Comment (tmacam) [inline tests/conformance/bindings/bindings.go:168]: Cool, but why not go one step further and split this into 2 tests: `ping-Input` and `ping-Output`? They could be run independently and one would be able to identify more easily what failed.

**Review by tmacam — APPROVED** [review]

File: tests/conformance/bindings/bindings.go [hunk 2]
```diff
@@ -143,23 +143,27 @@ func ConformanceTests(t *testing.T, props map[string]string, inputBinding bindin
...
 		}
-		errOut := bindings.PingOutBinding(outputBinding)
-		// TODO: Ideally, all stable components should implenment ping function,
-		// so will only assert assert.Nil(t, err) finally, i.e. when current implementation
-		// implements ping in existing stable components
-		if errOut != nil {
-			assert.EqualError(t, errOut, "ping is not implemented by this output binding")
-		} else {
-			assert.Nil(t, errOut)
+		if config.HasOperation("operations") {
```
Comment (ItalyPaleAle) [inline tests/conformance/bindings/bindings.go:157]: Is this a new operation? If so, please make sure you document it at the  top of [`tests/config/bindings/tests.yml`](https://github.com/dapr/components-contrib/blob/a4b27ae49b7c99820c6e921d3891f03334692714/tests/config/bindings/tests.yml#L1-L2)

Comment (ItalyPaleAle) [reply tests/conformance/bindings/bindings.go:157]: Please just document it in the bindings/test.yaml file then (and make sure that it's added to all test cases where it matters)

Comment (mcandeia) [reply tests/conformance/bindings/bindings.go:157]: Sorry @ItalyPaleAle, I’m not following you,

I did not create anything new, it’s just a matter of reusing of what existe and applying consistently with the initialization tests.

previously, the ping tests are executed regardless wether init was invoked or not. The operations that I’m using for this are the same for the lines above on the same file.

what do you mean for documenting those?

Comment (ItalyPaleAle) [reply tests/conformance/bindings/bindings.go:157]: You’re running that test if the operation `operations` is specified in the config/bindings/test.yml file. Unless you explicitly add that operation in the YAML, the branch with the test you added will not be executed.

Comment (mcandeia) [reply tests/conformance/bindings/bindings.go:157]: Do you know that the operation `operations` is not new, right?

If there are tests that had been executed because of `operations` is specified so they will continue to run.

And I don’t need to add “operations” for the tests that don’t declare anything because I’m fixing exactly the case where none operations are specified but ping tests were executed

Unless you meant to create a new operation “ping”, I guess we are fine.

Please, if you still thinking that something should be done, would you mind to point exactly where and what?

Comment (ItalyPaleAle) [reply tests/conformance/bindings/bindings.go:157]: > Do you know that the operation operations is not new, right?

No, I wasn't clear on that. I didn't see it in the YAML file documented so that's why I was asking "Is this a new operation". Thanks for confirming it's not a new one.

**Review by DeepanshuA — APPROVED** [review]
LGTM

**Review by berndverst — APPROVED** [review]
